File: Programmers/dnjfqhd12345/week_0/완주하지못한선수.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

string solution(vector<string> participant, vector<string> completion) {
    vector<string> vec(1);

    sort(participant.begin(), participant.end());
    sort(completion.begin(), completion.end());

    set_difference(participant.begin(), participant.end(), completion.begin(), completion.end(), vec.begin());
    string answer = vec[0];

    return answer;
}

File: Programmers/moonmonoon/week_4/완주하지못한선수.py
def solution(participant, completion):
    hash1 = {}
    hash2 = {}
    answer = ""

    for key in participant:
        if key in hash1:
            hash1[key] += 1
        else:
            hash1[key] = 1
    # print(hash1)

    for key in completion:
        if key in hash1:
            hash1[key] -= 1
    # print(hash1)

    for key in hash1:
        if hash1[key] != 0:
            answer = key

    return answer

# 100점!